### PR TITLE
Log request using zap

### PR DESCRIPTION
**Pull request by boekkooi-lengoo**
Good morning/evening/afternoon,

This PR add a custom LogFormatter to allow the request logs to log using zap instead of stdOut. \
This ensures that all logs are in the same format allows for better and easier parting by log aggregators.

This continues https://github.com/jhaals/yopass/pull/1082 and the log context is based on https://github.com/gorilla/handlers/blob/master/logging.go#L169-L185 to ensure that the same information is available.

Logs before:
```
10.10.0.107 - - [12/Nov/2021:09:36:09 +0000] "GET /locales/en.json HTTP/1.1" 200 4520
10.10.0.56 - - [12/Nov/2021:09:36:09 +0000] "GET /locales/en-US.json HTTP/1.1" 404 19
10.10.0.42 - - [12/Nov/2021:09:36:09 +0000] "GET /secret/3dba449f-3726-4255-9cea-d9b76f9831f2 HTTP/1.1" 200 243
```

After:
```json
{"level":"info","ts":1636712967.8815591,"caller":"server/logging.go:42","msg":"Request is handled","host":"127.0.0.1","username":"-","timestamp":1636712967.8807414,"method":"POST","uri":"/secret","path":"HTTP/1.1","responseStatus":200,"responseSize":50}
{"level":"info","ts":1636712972.5601518,"caller":"server/logging.go:42","msg":"Request is handled","host":"127.0.0.1","username":"-","timestamp":1636712972.5597835,"method":"GET","uri":"/secret/1c71469a-f99e-45bf-84d3-eace8a060bc2","path":"HTTP/1.1","responseStatus":200,"responseSize":259}
```

### CODE REVIEW

File: pkg/server/logging.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+func httpLogFormatter(logger *zap.Logger) func(io.Writer, handlers.LogFormatterParams) {
+	if logger == nil {
+		logger = zap.NewNop()
+	}
+
+	return func(_ io.Writer, params handlers.LogFormatterParams) {
+		var req = params.Request
+
+		username := "-"
```
Comment (jhaals) [inline pkg/server/logging.go:18]: Feels like user could be removed from the output

**Review by jhaals — COMMENTED** [review]
Thanks for looking into this!

File: pkg/server/logging.go (hunk 1 above)
Comment (boekkooi-lengoo) [reply pkg/server/logging.go:18]: You mean remove the entire `username` field or only add username field when it's set or just let it be an empty string?

Comment (jhaals) [reply pkg/server/logging.go:18]: Remove username entirely and in the output on line 45

Comment (boekkooi-lengoo) [reply pkg/server/logging.go]: Removed in https://github.com/jhaals/yopass/pull/1130/commits/8c76a4a0faee824f07058f370c4be492cfa74d32

**Comment by jhaals** [comment]
Nice 👍 Would it be possible to add a test for the formatter?

**Comment by boekkooi-lengoo** [comment]
> Nice +1 Would it be possible to add a test for the formatter?

I have added 2 test cases hope that is enough

**Comment by jhaals** [comment]
Nice!

**Pull request merged**